include/matx/core/tensor_utils.h

```diff
@@ -606,54 +622,165 @@ namespace detail {
    *
    */
   template <typename Op, typename ... Args>
-  __MATX_HOST__ void InternalPrint(const Op &op, Args ...dims)
+  __MATX_HOST__ void InternalPrint(FILE* fp, const Op &op, Args ...dims)
   {
     MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
 
     MATX_STATIC_ASSERT(op.Rank() == sizeof...(Args), "Number of dimensions to print must match tensor rank");
     MATX_STATIC_ASSERT(op.Rank() <= 4, "Printing is only supported on tensors of rank 4 or lower currently");
 
     if constexpr (sizeof...(Args) == 0) {
-      PrintVal(op.operator()());
-      printf("\n");
+      PrintVal(fp, op.operator()());
+      fprintf(fp, "\n");
     }
     else if constexpr (sizeof...(Args) == 1) {
       auto& k =detail:: pp_get<0>(dims...);
       for (index_t _k = 0; _k < ((k == 0) ? op.Size(0) : k); _k++) {
-        printf("%06" INDEX_T_FMT ": ", _k);
-        PrintVal(op.operator()(_k));
-        printf("\n");
+        if (_k == 0)
+        {
+          if ((PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_MLAB) || (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_PYTHON))
+          {
+            fprintf(fp, "[");
+          }
+        }
+        if (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_DEFAULT) fprintf(fp, "%06" INDEX_T_FMT ": ", _k);
+        PrintVal(fp, op.operator()(_k));
+        if (_k == (op.Size(0)-1))
+        {
+          if ((PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_MLAB) || (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_PYTHON))
+          {
+            fprintf(fp, "]");
+          }
+        }
+        else
+        {
+          if (PRINT_FORMAT_TYPE == MATX_PRINT_FORMAT_PYTHON)
+          {
+            fprintf(fp, ",");
```

Comment:
I think MATLAB allows commas between elements too. Can this be simplified by always adding it?

Reply:
Changed to always add for python and matlab